packages/wrangler/src/event-subscriptions/index.ts

```diff
@@ -0,0 +1,323 @@
+			return logger.table([
+				{ id: "", name: "", source: "", destination: "", events: "" },
+			]);
+		}
+
+		logger.table(
+			res.map((s) => ({
+				id: s.id,
+				name: s.name,
+				source: `${s.product_id}.${s.resource_id}`,
+				destination: `queues.${s.queue_name}`,
+				events: s.events.join(", "),
+			}))
+		);
+	},
+});
+
+export const eventSubscriptionsGetCommand = createCommand({
+	metadata: {
+		owner: "Product: Queues",
+		status: "open-beta",
+		description: "Fetch details about a single Event Subscription",
+	},
+
+	positionalArgs: ["id"],
+	args: {
+		id: {
+			type: "string",
+			description: "ID of the Event Subscription",
+			demandOption: true,
+		},
+	},
+
+	async handler(args, ctx) {
+		const config = readConfig(args);
+		const accountId = await requireAuth(config);
+		const { id } = args;
+
+		const subscription = await ctx.fetchResult<EventSubscription>(
+			`/accounts/${accountId}/event_subscriptions/subscriptions/${id}`
+		);
+
+		logger.table([
+			{
+				id: subscription.id,
+				name: subscription.name,
+				source: `${subscription.product_id}.${subscription.resource_id}`,
+				destination: `queues.${subscription.queue_name}`,
+				events: subscription.events.join(", "),
+			},
+		]);
+	},
+});
+
+export const eventSubscriptionsCreateCommand = createCommand({
+	metadata: {
+		owner: "Product: Queues",
+		status: "open-beta",
+		description:
+			"Create a new Event Subscription. Use `event-subscriptions browse` to see available services and events.",
+	},
+
+	positionalArgs: ["name"],
+	args: {
+		name: {
+			type: "string",
+			description: "",
+			demandOption: true,
+		},
+		source: {
+			type: "string",
+			description: "<Service>.<Resource> to subscribe to",
+			demandOption: true,
+		},
+		destination: {
+			type: "string",
+			description: "Name of the Queue to send events to",
+			demandOption: true,
+		},
+		events: {
+			type: "string",
+			description: "Names of events to subscribe to",
+			demandOption: true,
+		},
+	},
+	validateArgs({ source }) {
+		const [productId, resourceId] = source.split(".");
+
+		const product = PRODUCTS.find((p) => p.id === productId);
+		if (!product) {
+			throw new CommandLineArgsError(
+				`Invalid source. TODO: Write a better / more helpful error message`
+			);
+		}
+
+		if (product.resource) {
+			if (!resourceId) {
+				throw new CommandLineArgsError(
+					`Invalid resource ID. TODO: Write a better / more helpful error message`
+				);
+			}
+		}
+	},
+
+	async handler(args, ctx) {
+		const config = readConfig(args);
+		const accountId = await requireAuth(config);
+
+		const { name, source, destination: queueId, events } = args;
+
+		const [productId, resourceId] = source.split(".");
+		const product = PRODUCTS.find((p) => p.id === productId);
+
+		// Disabling ESLint and using `!` because we've validated the argument already
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const res = await product!.subscription.create(ctx, {
```

Comment:
> using `!` because we've validated the argument already

What guarantees that `PRODUCTS.find((p) => p.id === productId)` will always return a non-null result?

Reply:
So up in the `validateArgs` function, I do the same lookup and throw an error if the product wasn't found. This code path would be unreachable if the product did not exist.